packages/features/webhooks/lib/subscriberUrlReserved.ts

```diff
@@ -0,0 +1,39 @@
+import type { Webhook } from "@calcom/prisma/client";
+
+interface Params {
+  subscriberUrl: string;
+  id?: string;
+  webhooks?: Webhook[];
+  teamId?: number;
+  userId?: number;
+  eventTypeId?: number;
+}
+
+export const subscriberUrlReserved = ({
+  subscriberUrl,
+  id,
+  webhooks,
+  teamId,
+  userId,
+  eventTypeId,
+}: Params): boolean => {
+  if (teamId) {
+    return !!webhooks?.find(
+      (webhook) =>
+        webhook.subscriberUrl === subscriberUrl && (!id || webhook.id !== id) && webhook.teamId === teamId
+    );
+  }
+  if (eventTypeId) {
+    return !!webhooks?.find(
+      (webhook) =>
+        webhook.subscriberUrl === subscriberUrl &&
+        (!id || webhook.id !== id) &&
+        webhook.eventTypeId === eventTypeId
+    );
+  } else {
+    return !!webhooks?.find(
+      (webhook) =>
+        webhook.subscriberUrl === subscriberUrl && (!id || webhook.id !== id) && webhook.userId === userId
+    );
+  }
```

Comment:
```suggestion
const findMatchingWebhook = (condition) => {
  return !!webhooks?.find(
    (webhook) =>
      webhook.subscriberUrl === subscriberUrl &&
      (!id || webhook.id !== id) &&
      condition(webhook)
  );
};

if (teamId) {
  return findMatchingWebhook((webhook) => webhook.teamId === teamId);
}

if (eventTypeId) {
  return findMatchingWebhook((webhook) => webhook.eventTypeId === eventTypeId);
}

return findMatchingWebhook((webhook) => webhook.userId === userId);
```
NIT: Reusing logic

Reply:
I would also suggest adding a runtime check for teamId or eventTypeId to be present atleast. I don't see a reason, why both of them can be nullish.